packages/frontend/src/state/wallet/hooks.ts

```diff
@@ -0,0 +1,233 @@
+import { useAtom } from 'jotai'
+import { useUpdateAtom } from 'jotai/utils'
+import BigNumber from 'bignumber.js'
+import Notify from 'bnc-notify'
+import Onboard from 'bnc-onboard'
+import Web3 from 'web3'
+import { ethers } from 'ethers'
+import { useQuery, useQueryClient } from 'react-query'
+
+import { onboardAtom, addressAtom, notifyAtom, networkIdAtom, signerAtom, web3Atom } from './atoms'
+import { BIG_ZERO, EtherscanPrefix } from '../../constants/'
+import { Networks } from '../../types'
+import { useCallback, useEffect } from 'react'
+
+export const useSelectWallet = () => {
+  const [onboard] = useAtom(onboardAtom)
+  const onWalletSelect = async () => {
+    if (!onboard) return
+    onboard.walletSelect().then(async (success) => {
+      if (success) await onboard.walletCheck()
+    })
+  }
+
+  return onWalletSelect
+}
+
+export const useDiscconectWallet = () => {
+  const [onboard] = useAtom(onboardAtom)
+  const setAddress = useUpdateAtom(addressAtom)
+  const queryClient = useQueryClient()
+  //   const setBalance = useUpdateAtom(balanceAtom)
+  const disconnectWallet = () => {
+    if (!onboard) return
+    onboard.walletReset()
+    setAddress(null)
+    queryClient.setQueryData('userWalletBalance', BIG_ZERO)
+    // setBalance(BIG_ZERO)
+  }
+
+  return disconnectWallet
+}
+
+export const useHandleTransaction = () => {
```

Comment:
Could we maybe make tx related info or tx receipt (i.e. chainId, TransactionType, TransactionHash, blockNumber, address from, address to...etc) & tx action(tx sent, tx finalize...etc) into a new Atom object? Which would be a global state and easier for testing to track all actions after users send txs

Reply:
Okay, I will see what can be done.

Reply:
Oh thanks